muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/WorkspaceBackingBean.java

```diff
@@ -79,24 +79,28 @@ public WorkspaceEntity getWorkspaceEntity() {
   }
 
   public void setWorkspaceUrlName(String workspaceUrlName) {
+    
+    WorkspaceUserEntity workspaceUserEntity = null;
     WorkspaceEntity workspaceEntity = resolveWorkspaceEntity(workspaceUrlName);
     if (workspaceEntity != null) {
       this.workspaceEntityId = workspaceEntity.getId();
       this.workspaceUrlName = workspaceEntity.getUrlName();
+      if (sessionController.isLoggedIn()) {
+        workspaceUserEntity = workspaceUserEntityController.findWorkspaceUserByWorkspaceEntityAndUserIdentifier(workspaceEntity, sessionController.getLoggedUser());
+      }
     }
+    boolean activeWorkspaceUser = workspaceUserEntity != null && Boolean.TRUE.equals(workspaceUserEntity.getActive());
+    
     homeVisible = workspaceToolSettingsController.getToolVisible(workspaceEntity, "home");
     guidesVisible = workspaceToolSettingsController.getToolVisible(workspaceEntity, "guides");
     materialsVisible = workspaceToolSettingsController.getToolVisible(workspaceEntity, "materials");
-    discussionsVisible = sessionController.hasWorkspacePermission(ForumResourcePermissionCollection.FORUM_ACCESSWORKSPACEFORUMS, workspaceEntity) && workspaceToolSettingsController.getToolVisible(workspaceEntity, "discussions");
-    usersVisible = sessionController.hasWorkspacePermission(MuikkuPermissions.MANAGE_WORKSPACE_MEMBERS, workspaceEntity) && workspaceToolSettingsController.getToolVisible(workspaceEntity, "users");
-    journalVisible = sessionController.hasWorkspacePermission(MuikkuPermissions.ACCESS_WORKSPACE_JOURNAL, workspaceEntity) && workspaceToolSettingsController.getToolVisible(workspaceEntity, "journal");
+    discussionsVisible = activeWorkspaceUser && sessionController.hasWorkspacePermission(ForumResourcePermissionCollection.FORUM_ACCESSWORKSPACEFORUMS, workspaceEntity) && workspaceToolSettingsController.getToolVisible(workspaceEntity, "discussions");
```

Comment:
Doesn't work for administrative users as they fulfill the permission check even when not being workspace users...